lib/src/stream_subscription_transformer.dart

```diff
@@ -18,10 +18,14 @@ typedef void _SimpleHandler<T>(StreamSubscription<T> inner);
 /// [StreamSubscription.resume] is called, the corresponding handler is invoked.
 /// By default, handlers just forward to the underlying subscription.
 ///
-/// This guarantees that once the transformed [StreamSubscription] has been
-/// canceled, no further callbacks will be invoked. The [handlePause] and
-/// [handleResume] are invoked regardless of whether the subscription is paused
-/// already or not.
+/// Guarantees that once the transformed [StreamSubscription] has been canceled,
+/// no further callbacks will be invoked. The [handlePause] and [handleResume]
+/// are invoked regardless of whether the subscription is paused already or not.
+///
+/// In order to preserve [StreamSubscription] guarantees, **all callbacks must
+/// synchronously call the corresponding method** on the inner
```

Comment:
I could imagine a user wanting to add some behavior after the `cancel()` future completes. I suppose we could make `handlePause`/`handleResume` notifications, but that would be oddly inconsistent.